### PR TITLE
Fix packages broken by gcc 10->11 update

### PR DESCRIPTION
**Pull request by balsoft**
###### Description of changes

Attempt to (automatically) fix all packages requiring gcc10Stdenv.

Script:

```
#!/usr/bin/env nix-shell
#!nix-shell -p jq moreutils -i bash

export PATH=/home/balsoft/.local/share/cargo/bin:"$PATH"

nixpkgs="${nixpkgs-/home/balsoft/projects/nixpkgs}"

all_packages="$nixpkgs/pkgs/top-level/all-packages.nix"

patchPackage() {
    ( set -euo pipefail
      initial="$(nix-editor $all_packages "$1")"
      new="$(printf '%s' "$initial" | cut -d' ' -f1,2) { stdenv = gcc10StdenvCompat; $(printf '%s' "$initial" | tr '\n' ' ' | cut -d' ' -f4-)"
      nix-editor "$all_packages" -v "$new" "$1" | sed '$ d' | sponge "$all_packages"
    )
}

cd "$nixpkgs"

for i in $(cat ~/oof.json | jq '.[]' -r | rev | cut -d. -f2- | rev | grep -v "\."); do
    printf "%s" "$i"
    patchPackage $i
    if ! timeout --preserve-status 5 sh -c "nix-env --option system x86_64-linux -f $nixpkgs -qaP --xml --out-path --show-trace --meta || echo 1" > /dev/null; then
        printf " : success!\n"
        git add --all
        git commit -m "$i: stdenv -> gcc10Stdenv" --no-gpg-sign
    else
        printf " : failure!\n"
        git reset --hard HEAD
    fi
done
```

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mindavi** [comment]
What does this do when stdenv is still at gcc9, like for aarch64-linux?

**Comment by balsoft** [comment]
```
25 packages failed to build:
clingcon djv dl-poly-classic-mpi dxx-rebirth freeswitch gammy getdp hobbes mars miniaudicle mrustc-bootstrap nano-wallet openvino oxen pianobooster python310Packages.openvino python39Packages.openvino rss-glx sonic-lineup spring springLobby stuntrally tracebox webrtc-audio-processing_1 yafaray-core

42 packages built:
cxxtools dbxml diopser ericw-tools fluxus flwrap gcc10StdenvCompat gebaar-libinput gosmore gsmlib idsk iqueue jigdo libdynd pktgen loki mitscheme mitschemeX11 mkcue mps non obliv-c openclonk opendbx percona-server percona-xtrabackup percona-xtrabackup_2_4 python310Packages.dynd python39Packages.dynd strelka swfmill swiftshader textadept11 tiscamera tntdb tntnet tora ucg uri wxcam xc3sprog xqilla
```

Therse packages couldn't be fixed-up automatically: cemu cntk copper d-seams grub4dos gscrabble gst_all_1.gstreamermm halide ifcopenshell klee ksmoothdock libsForQt514.qt5.qtwebengine libsForQt514.qtwebengine linuxKernel.kernels.linux_4_14_hardened linuxKernel.kernels.linux_mptcp_95 linuxKernel.packages.hardkernel_4_14.pktgen linuxKernel.packages.linux_4_14.pktgen linuxKernel.packages.linux_4_14_hardened.pktgen linuxKernel.packages.linux_4_19.pktgen linuxKernel.packages.linux_4_19_hardened.pktgen linuxKernel.packages.linux_4_9.pktgen linuxKernel.packages.linux_5_10.pktgen linuxKernel.packages.linux_5_10_hardened.pktgen linuxKernel.packages.linux_5_15.pktgen linuxKernel.packages.linux_5_15_hardened.pktgen linuxKernel.packages.linux_5_17.pktgen linuxKernel.packages.linux_5_4.pktgen linuxKernel.packages.linux_5_4_hardened.pktgen linuxKernel.packages.linux_hardened.pktgen linuxKernel.packages.linux_latest_libre.pktgen linuxKernel.packages.linux_libre.pktgen linuxKernel.packages.linux_lqx.pktgen linuxKernel.packages.linux_testing_bcachefs.pktgen linuxKernel.packages.linux_xanmod.pktgen linuxKernel.packages.linux_zen.pktgen linuxPackages-libre.pktgen linuxPackages.pktgen linuxPackages_4_14_hardened.pktgen linuxPackages_4_19_hardened.pktgen linuxPackages_5_10_hardened.pktgen linuxPackages_5_15_hardened.pktgen linuxPackages_5_4_hardened.pktgen linuxPackages_hardened.pktgen linuxPackages_hardkernel_latest.pktgen linuxPackages_latest-libre.pktgen linuxPackages_latest.pktgen linuxPackages_lqx.pktgen linuxPackages_testing_bcachefs.pktgen linuxPackages_xanmod.pktgen linuxPackages_zen.pktgen linux_mptcp mitschemeX11 mozart2 mythtv onioncircuits partio perseus pulseview python310Packages.graphite_api python310Packages.ifcopenshell python310Packages.openvino python310Packages.pyramid_jinja2 python310Packages.softlayer python310Packages.sphinx-markdown-parser python310Packages.torchinfo python310Packages.wasmer-compiler-llvm python310Packages.wasmerPackages.wasmer-compiler-llvm python39Packages.graphite_api python39Packages.ifcopenshell python39Packages.modeled python39Packages.openvino python39Packages.pymc3 python39Packages.pyramid_jinja2 python39Packages.softlayer python39Packages.sphinx-markdown-parser python39Packages.torchinfo python39Packages.wasmer-compiler-llvm python39Packages.wasmerPackages.wasmer-compiler-llvm qt514.qtwebengine retroshare sta xorg.xf86videoglint xorg.xf86videosuncg6 xorg.xf86videosunffb xorg.xf86videosunleo xorg.xf86videotrident

**Comment by balsoft** [comment]
> What does this do when stdenv is still at gcc9, like for aarch64-linux?

Good question. This indeed should be handled!

**Review by cab404 — COMMENTED** [review]
lgtm, let’s goooo

**Review by cab404 — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -32284,12 +32269,8 @@ with pkgs;
...
   stt = callPackage ../tools/audio/stt { };
 
-  stuntrally = callPackage ../games/stuntrally {
-    ogre = ogre1_9;
-    mygui = mygui.override {
-      withOgre = true;
-    };
-  };
+  stuntrally = callPackage ../games/stuntrally
+    { stdenv = gcc10StdenvCompat; ogre = ogre1_9; mygui = mygui.override { withOgre = true; }; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:32273]: This formatting is very odd.

```suggestion
;
  stuntrally = callPackage ../games/stuntrally {
    stdenv = gcc10StdenvCompat;
    ogre = ogre1_9;
    mygui = mygui.override { withOgre = true; };
  };
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -6732,9 +6732,8 @@ with pkgs;
 
   gsmartcontrol = callPackage ../tools/misc/gsmartcontrol { };
 
-  gsmlib = callPackage ../development/libraries/gsmlib {
-    autoreconfHook = buildPackages.autoreconfHook269;
-  };
+  gsmlib = callPackage ../development/libraries/gsmlib
+    { stdenv = gcc10StdenvCompat; autoreconfHook = buildPackages.autoreconfHook269; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:6736]:
```suggestion
  gsmlib = callPackage ../development/libraries/gsmlib {
    stdenv = gcc10StdenvCompat;
    autoreconfHook = buildPackages.autoreconfHook269;
  };
```

**Pull request merged**